quartz/components/scripts/popover.inline.ts

```diff
@@ -37,29 +37,47 @@ async function mouseEnterHandler(
   targetUrl.hash = ""
   targetUrl.search = ""
 
-  const contents = await fetch(`${targetUrl}`)
-    .then((res) => res.text())
-    .catch((err) => {
-      console.error(err)
-    })
+  const response = await fetch(`${targetUrl}`).catch((err) => {
+    console.error(err)
+  })
 
   // bailout if another popover exists
   if (hasAlreadyBeenFetched()) {
     return
   }
 
-  if (!contents) return
-  const html = p.parseFromString(contents, "text/html")
-  normalizeRelativeURLs(html, targetUrl)
-  const elts = [...html.getElementsByClassName("popover-hint")]
-  if (elts.length === 0) return
+  if (!response) return
+  const contentType = response.headers.get("Content-Type")
 
   const popoverElement = document.createElement("div")
   popoverElement.classList.add("popover")
   const popoverInner = document.createElement("div")
   popoverInner.classList.add("popover-inner")
   popoverElement.appendChild(popoverInner)
-  elts.forEach((elt) => popoverInner.appendChild(elt))
+
+  const contentTypeCategory = contentType?.split("/")[0] ?? "text"
+  switch (contentTypeCategory) {
+    case "image":
+      const img = document.createElement("img")
+
+      response.blob().then((blob) => {
+        img.src = URL.createObjectURL(blob)
+      })
+      img.alt = targetUrl.pathname
+
+      popoverInner.appendChild(img)
+      popoverInner.classList.add("popover-inner-img")
```

Comment:
Hmm I think we can save a dataset field like `imagePopover` to true then use CSS matching for `data-image-popover`?

Probably don't need to get a class here.

Reply:
Made it a bit more general by setting `data-content-type` to whatever the `contentTypeCategory` is. Unsure if video popovers are supported but this would leave it more open for future changes. Also just checked; with the current implementation SVGs work fine as well
<img width="580" alt="image" src="https://github.com/jackyzha0/quartz/assets/33621804/95500a3f-b658-45c3-9f23-295e107f68f4">